certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java

```diff
@@ -28,33 +29,26 @@
 
     private static Logger logger = LoggerFactory.getLogger(CertificateFactory.class);
 
-    final String resourceName = "application.properties";
 
-    private static SignatureHelper signatureHelper;
+    private ObjectMapper mapper = new ObjectMapper();
 
-    public CertificateExtension createCertificate(CertModel certModel, Map<String, String> properties) throws InvalidDateFormatException {
+    public CertificateExtension createCertificate(CertModel certModel, Map<String, String> properties, String keyID) throws InvalidDateFormatException {
 
-        uuid = properties.get(JsonKey.DOMAIN_URL).concat("/")+ UUID.randomUUID().toString();
+        uuid = properties.get(JsonKey.DOMAIN_URL).concat("/") + properties.get(JsonKey.CONTAINER_NAME).concat("/")
+                + properties.get(JsonKey.ROOT_ORG_ID).concat("/") + properties.get(JsonKey.TAG).concat("/") + UUID.randomUUID().toString() + ".json";
 
         CertificateExtensionBuilder certificateExtensionBuilder = new CertificateExtensionBuilder(properties.get(JsonKey.CONTEXT));
         CompositeIdentityObjectBuilder compositeIdentityObjectBuilder = new CompositeIdentityObjectBuilder(properties.get(JsonKey.CONTEXT));
         BadgeClassBuilder badgeClassBuilder = new BadgeClassBuilder(properties.get(JsonKey.CONTEXT));
         IssuerBuilder issuerBuilder = new IssuerBuilder(properties.get(JsonKey.CONTEXT));
         SignedVerification signedVerification = new SignedVerification();
-
+        SignatureBuilder signatureBuilder = new SignatureBuilder();
 
         Criteria criteria = new Criteria();
         criteria.setNarrative("For exhibiting outstanding performance");
```

Comment:
"For exhibiting outstanding performance" value can't be hardcoded .
where this value is shown?